Unable to create Payment

fernando-barajas opened the pull request:
When an user wants to pay and order that contains an 'Free Shipping'
promotion, the charge is rejected with the error 'Handling Total is
Invalid'.
- Include the shipping and tax adjustments for the amount of credit_total in order to avoid a negative handling for order_opts
- Fix "Checkout" test

Commit:
Unable to create Payment

When an user wants to pay and order that contains an 'Free Shipping'
promotion, the charge is rejected with the error 'Handling Total is
Invalid'. In order to prevent this issue we're removing the shipping
total from the data sent to Paypal.

- Add logic to check if the order has a 'Free Shipping' promotion
- Remove the cost of the shipping if the order has a 'Free Shipping'
  promotion

fernando-barajas commented:
@chischaschos @hecbuma could you review this PR?

Commit:
- Use a where clause instead of a select to check if the order has a "Free Shipping" promotion

chischaschos commented:
This looks good, do you know what is the status of the test suite?

fernando-barajas commented:
@chischaschos the test suite is broken, I could spend some time to fix it

joksovicn commented:
In my opinion, this should go back to the main project. The spree_paypal_express gem should not have any dependency to our "free shipping", since this could change at any time.

chischaschos commented:
@joksovicn sound good, hadn't thought about this. I agree if this logic is added here then this gem would have dependencies on our LB spree app, it should be the other way.

@fernando-barajas, @hecbuma and I talked about it, @fernando-barajas will move the logic back to the LB spree app.

Commit:
ERP-380 - LB - Bug with checkout when PayPal is payment method

When an user wants to pay and order that contains an 'Free Shipping'
promotion, the charge is rejected with the error 'Handling Total is
Invalid'. In order to prevent this issue we're removing the shipping
total from the data sent to Paypal

- Get all adjustments of the order including line items, shipments and
  taxes in order to get a proper subtotal and avoid a negative handling
  option
- Fix checkout tests

fernando-barajas commented:
@chischaschos @joksovicn could you review this PR?

Commit:
Fix PR comments

chischaschos commented:
LGTM!, we will add this project to circle ci tomorrow

Commit:
- Fix issue when creating dummy app due to invalid spree testing support path

Commit:
ERP-380 Enable CI

- Enable CI with circle ci
- Fix dummy app paths
- Add coffee_rails dependency as in master branch
- Allow generator to receive environment variable to run migrations on
  CI

chischaschos commented:
Circle CI is failling but all the checkout specs, which are the ones that @fernando-barajas fixed are green. So LGTM! thank you @fernando-barajas .

Commit:
ERP-380 Enable CI

- Enable CI with circle ci
- Fix dummy app paths
- Add coffee_rails dependency as in master branch
- Allow generator to receive environment variable to run migrations
  on CI
